### PR TITLE
Feature/working branch

### PR DESCRIPTION
**Pull request by Maryjanee**
Following the completion of the project, the authors request the merge from the feature/working_branch to the development branch.

   The following was implemented in the project:
- Users can create an account with their just a username
- Users can create events 
- Users can see past events and upcoming ones
- Users can see all events created

### CODE REVIEW

File: db/migrate/20200916100344_create_events.rb [hunk 1]
```diff
@@ -0,0 +1,9 @@
+rclass CreateEvents < ActiveRecord::Migration[6.0]
```
Comment (christianotieno) [inline db/migrate/20200916100344_create_events.rb]: # Fix the typo error
- [ ] Remove the unnecessary character at the beginning of the `CreateEvents` class definition. It is causing migration errors for users of your code, thank you!
```suggestion
class CreateEvents < ActiveRecord::Migration[6.0]
```

File: config/routes.rb [hunk 2]
```diff
@@ -0,0 +1,12 @@
+Rails.application.routes.draw do
+  # For details on the DSL available within this file, see https://guides.rubyonrails.org/routing.html
+  resources :users, only: %i[new create show]
+  get 'sessions/new', to: 'sessions#new' 
+  post 'sessions', to: 'sessions#create'
+  delete 'sessions', to: 'sessions#destroy'
```
Comment (christianotieno) [inline config/routes.rb:6]: # OPTIONAL CHANGES ♻️ 
- [x] Refactoring this code to have at least one root route will be an awesome User Experience.

File: config/routes.rb [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+  # EventsController routes
+  get 'events/new', to: 'events#new'
+  post 'events/new', to: 'events#create'
+  get 'events', to: 'events#index'
+  get 'events/:id', to: 'events#show'
```
Comment (christianotieno) [inline config/routes.rb]: # OPTIONAL CHANGES ♻️ 
- [ ] Refactoring this lines of code to the following will be neat.
```suggestion
  resources :events, only: [:new, :create, :show, :index]
```

File: config/routes.rb [hunk 4]
```diff
@@ -0,0 +1,12 @@
...
+end
```
Comment (christianotieno) [inline config/routes.rb:10]: # OPTIONAL CHANGES ♻️ 

### Invites feature \[EXTRA CREDIT IN ODIN\]

* [ ] route under `/event/:event_id/invites/new`
* [ ] Invite model `belongs_to :event`
* [ ] Invite model `belongs_to :user`
* [ ] User model `has_many :invitations`
* [ ] Event model `has_many :invitations`
* [ ] Join model (EventAttendance) is required to make sure that atenndance is not created without invitation (e.g. `before_create` callback usage)

**Review by christianotieno — CHANGES_REQUESTED** [review]
## CHANGES REQUIRED ♻️ 

Hello team 👋 .  Good job with the project. keep it up.
Let's take care of the suggestions below 👇 before we continue.

#### Christian Otieno (TSE)

File: config/routes.rb (hunk 2 above)
Comment (Joseph-Burke) [reply config/routes.rb:6]: A root route has now been created, directing users to events#index

**Comment by Joseph-Burke** [comment]
All changes required by @christianotieno have now been implemented, and all but one of the optional changes suggested have also been implemented.

The "Extra Credit" invitation feature requires broader and more significant changes to the project so @Maryjanee and I have decided to make this change after project approval.

**Review by rOluochKe — APPROVED** [review]
Hello @Maryjanee and @Maryjanee 
### Well done, you got it right :clap: The project has been approved. Congratulations :boom: :100:
### Merge and submit the project. Good luck with the net one.
- Happy coding!

**Pull request merged**